### ISSUE TITLE
How do I fill in the .json files?

### ISSUE THREAD

**Issue by Ecdldaiiere** [Coombaa/TTAutoRecord#52]
It's not clear to me exactly how I should fill out the live_users.json and/or the stream_links.json so that I can start recording users. I have already filled in the cookies.json. 

I tried putting the .flv stream link in between the curly brackets in stream_links.json but that didn't work. For the live_users.json, I tried to simply just put the username on one line but that didn't work either. 

Currently when I start the main.py, I keep getting messages saying "JSON file is empty". Can someone be a little bit more clear on how I can add users to start recording? The readme does not explain any of this. Thanks.

**Comment by FabLeKebab** [comment]
Hi, let’s start from the beginning:

I assume everything is installed properly, along with all the required modules. The issue seems to be with the cookies. Are your cookies in a "list" format or a big "block"?

If it’s a "block" without any structure, I recommend installing Cookie-Editor. It works perfectly for me.

If that’s not the case, did you set TikTok to English before retrieving the cookies? And were you on the page https://www.tiktok.com/live? (You need to retrieve the cookies specifically from this page while logged in.)


If none of this works, delete everything and start over from scratch:

1. Install Python 3.10.11 (this version works perfectly for me across all applications).


2. Install the **Cookie-Editor** extension on Chrome (it’s better than the other one, which outputs a "blob" of data).


3. Download and install **Mozilla Firefox** (you don’t need to log in, just install it).


4. In the repository, locate the file **requirements.txt** and run this command from that folder:

```
pip install -r requirements.txt
```


5. On Chrome, log in to TikTok. **Set it to English** before exporting the cookies and ensure you’re on https://www.tiktok.com/live.


6. Once logged in, on this page, in English, open the **Cookie-Editor** extension. Click **Export**, then select "JSON". The cookies will be copied to your clipboard.


7. Paste these cookies into the file `\json\cookies.json`.



Finally, run the **main.py** file. (Although, the GUI can be a bit annoying as it opens two windows every time...)

Let me know how it goes!

**Comment by Ecdldaiiere** [comment]
I used Cookie-Editor extension, exported the json and pasted into cookies.json. And yes I did get cookies from the tiktok.com/live page and yes it is set to English. When I start main.py though, I still get message saying JSON file is empty along with a popup saying "Currently recording 0/0". 

My question is, am I supposed to add the users I want to record **AFTER** I run the main.py or am I supposed to add the users I want to record into the live_users.json and/or the stream_links.json. If I am supposed to add the users I want to record into the .json, how do I fill in those .jsons? What exactly am I supposed to put into those .json files?

This is a screenshot of what messages I'm getting: https://i.imgur.com/r3GGbCw.png

**Comment by FabLeKebab** [comment]
No, there’s nothing more to do. You simply export the cookies with the extension, and once that’s done, the live user downloads start automatically. Other than installing Firefox if it’s not already installed, there’s nothing else. Simply put, the cookies are used to retrieve users, and as soon as someone goes live, the recording starts about 10 seconds later. If you’re getting this error message, try reinstalling as explained above. Are you sure you’re using the "main" repository?

**Comment by Ecdldaiiere** [comment]
So how does it know which users to retrieve? Does it retrieve only the Lives of people you are following? Here is a screenshot https://i.imgur.com/r3GGbCw.png

And this is another screenshot after I cleared the stream_links.json : https://i.imgur.com/HyoPCw6.png

**Comment by FabLeKebab** [comment]
Whoa, I think this is the first version I used, so it’s probably not the right one (or maybe I don’t have the right colors), but oh well, never mind. **To answer your question**, it only records **the people you are subscribed to**. Once you’re logged in, you don’t need to do anything else, just leave your TikTok account alone and don’t delete the cookies (otherwise it will log you out and you’ll have to re-add the cookies).

Once you’re logged into your account, just subscribe to people and they will automatically be added to the downloads. Try going through the steps I mentioned earlier with this repository: https://github.com/Coombaa/TTAutoRecord/archive/refs/heads/main.zip

Effectively, you're not using the right version; you took version 4.1.2, which has been outdated for a long time. With the link I sent you, everything should work. Let me know how it goes. Once everything is set up and the download starts, leave everything as it is and don't touch anything. Keep your TikTok connected to Chrome (you can close Chrome, there's no need to leave TikTok open in the background). Just make sure your PC stays on, and check from time to time that your TikTok account remains logged in on Chrome.

**Comment by Ecdldaiiere** [comment]
So I did everything exactly as you said and it looks a little bit different but still won't record the live user I'm followed to. Now it looks like this. https://i.imgur.com/mBCer7I.png 

Seems to be making some progress though because now as you said, the live_users.json automatically populated with data of the user I'm trying to record. Just not sure why it's saying no stream link found even though they are currently live.

**Comment by FabLeKebab** [comment]
Yeah, I had the same issue with my old account, which is why I wrote this here: https://github.com/Coombaa/TTAutoRecord/issues/51. I tried logging into an old account, subscribing to a few people, and it started working perfectly. If you have an old account, try using that one. If it works, you can gradually re-subscribe to the accounts you follow (if you don't follow too many, it should be fine)

**Comment by FabLeKebab** [comment]
However, I’ve noticed that the error messages are different from mine... It's strange, as I’m using the exact same repository. Personally, I use it on two different PCs, and both are working fine (with two separate accounts as well, the one that’s “down” and the other one I took over to re-subscribe)

![image](https://github.com/user-attachments/assets/b30c8a8a-e43b-4348-a124-6e52df57426a)
![image](https://github.com/user-attachments/assets/26d2c69d-fd6a-41c5-bce1-176a5b7bde00)

**Comment by Ecdldaiiere** [comment]
When you say "subscribe" do you actually mean "subscribing" like paying? Or do you just mean follow? Because they are two separate things on TikTok. I tried unfollowing and refollowing the user I'm trying to get the livestream of but all I get is a message saying:

No matching stream link found in the page source
No stream link found for (user)

**Comment by FabLeKebab** [comment]
> When you say "subscribe" do you actually mean "subscribing" like paying? Or do you just mean follow? Because they are two separate things on TikTok. I tried unfollowing and refollowing the user I'm trying to get the livestream of but all I get is a message saying:
> 
> No matching stream link found in the page source
> No stream link found for (user)

I was indeed referring to the free subscription, not the paid one. That being said, if none of the solutions work on your end, unfortunately, there's nothing more I can do. Everything works perfectly on my side, with two different accounts on two different computers, using the same repository and exactly the same instructions. So I really don’t understand where the issue is coming from... :\

**Comment by shadowdragon245** [comment]
@Ecdldaiiere if you want dm me on telegram https://t.me/shadowdragon245 i can try and help + i have a vesion to try
